sdk/include/string.h

```diff
@@ -6,27 +6,37 @@
 #include <stddef.h>
 #include <stdint.h>
 
-int __cheri_libcall    memcmp(const void *str1,
-                              const void *str2,
-                              size_t      count) __asm__("memcmp");
-void *__cheri_libcall  memcpy(void       *dest,
-                              const void *src,
-                              size_t      n) __asm__("memcpy");
-void *__cheri_libcall  memset(void *, int, size_t) __asm__("memset");
-void *__cheri_libcall  memmove(void       *dest,
-                               const void *src,
-                               size_t      n) __asm__("memmove");
-void *__cheri_libcall  memchr(const void *, int, size_t);
-void *__cheri_libcall  memrchr(const void *, int, size_t);
-size_t __cheri_libcall strlen(const char *str);
-int __cheri_libcall    strncmp(const char *s1, const char *s2, size_t n);
-char *__cheri_libcall  strncpy(char *dest, const char *src, size_t n);
-int __cheri_libcall    strcmp(const char *s1, const char *s2);
-char *__cheri_libcall  strnstr(const char *haystack,
-                               const char *needle,
-                               size_t      haystackLength);
-char *__cheri_libcall  strchr(const char *s, int c);
-size_t __cheri_libcall strlcpy(char *dest, const char *src, size_t n);
+#define DECLARE_STRING_LIBCALL(rtype, name, ...)                               \
```

Comment:
It would be good to have a comment here for what this is doing (declaring libcall functions with unmangled names) and why it exists (to permit compiler optimisations).

Minor naming nit, but are all of the things where we care about compiler libcalls optimisations string functions?  If not, we should maybe call this `DECLARE_STANDARD_LIBCALL` or similar.